tests/foreman/cli/test_variables.py

```diff
@@ -0,0 +1,1120 @@
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher_merge_override(self):
+        """Merge the values of all the associated matchers.
+
+        @id: b9c9b1c7-ff9a-4080-aeee-3b61b5414332
+
+        @steps:
+
+        1.  Create variable with some default value.
+        2.  Create first matcher for attribute fqdn with valid details.
+        3.  Create second matcher for other attribute with valid details.
+        Note - The fqdn/host should have this attribute.
+        4.  Create more matchers for some more attributes if any.
+        Note - The fqdn/host should have this attributes.
+        5.  Set --merge-overrides to true.
+        6.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values merged from all the associated
+        matchers.
+        2.  The YAML output doesn't have the default value of variable.
+        3.  Duplicate values in YAML output if any are displayed.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create_matcher_merge_override(self):
+        """Test merge the values from non associated matchers.
+
+        @id: 3cc2a7b3-7b46-4c8c-b719-79c004ae04c6
+
+        @steps:
+
+        1.  Create variable with some default value.
+        3.  Create first matcher for attribute fqdn with valid details.
+        4.  Create second matcher for other attribute with valid details.
+        Note - The fqdn/host should not have this attribute.
+        5.  Create more matchers for some more attributes if any.
+        Note - The fqdn/host should not have this attributes.
+        6.  Set --merge-overrides to true.
+        7.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values only for fqdn.
+        2.  The YAML output doesn't have the values for attribute
+        which are not associated to host.
+        3.  The YAML output doesn't have the default value of variable.
+        4.  Duplicate values in YAML output if any are displayed.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher_merge_default(self):
+        """Merge the values of all the associated matchers + default value.
+
+        @id: 21d8fde8-0844-4384-b86a-30547c82b221
+
+        @steps:
+
+        1. Create variable with some default value.
+        2. Create first matcher for attribute fqdn with valid details.
+        3. Create second matcher for other attribute with valid details.
+        Note - The fqdn/host should have this attribute.
+        4. Create more matchers for some more attributes if any.
+        Note - The fqdn/host should have this attributes.
+        5. Set --merge-overrides to true.
+        6. Set --merge-default to true.
+        7. Go to YAML output of associated host.
```

Comment:
We don't have yaml output in CLI, even if you specify --output-format=yaml, it will not be the same as we have for UI